packages/account/src/Sections/Security/Passkeys/passkeys.tsx

```diff
@@ -1,144 +1,205 @@
-    }, [error, is_passkey_registration_started]);
+        return () => clearTimeOut(error_modal_timeout);
+    }, [error, is_reminder_modal_open]);
 
-    if (should_show_passkeys && (is_passkeys_list_loading || common.network_status.class !== 'online')) {
+    if (should_show_passkeys && (is_passkeys_list_loading || !is_network_on)) {
         return <Loading is_fullscreen={false} className='account__initial-loader' />;
     }
+
     if (!should_show_passkeys) {
         return <Redirect to={routes.traders_hub} />;
     }
 
-    // to avoid flickering with blank Modal we need first close it, then clear content
-    const onCloseModal = (delayed_action: () => void) => {
-        setIsModalOpen(false);
-        if (timeout) clearTimeout(timeout);
-        timeout = setTimeout(delayed_action, 300);
-    };
-
-    const onCloseError = () => {
+    const onCloseErrorModal = () => {
         if (passkey_registration_error) {
             clearPasskeyRegistrationError();
         }
+        history.push(routes.traders_hub);
     };
-    const onModalButtonClick = () => {
-        if (error) {
-            onCloseModal(onCloseError);
-            history.push(routes.traders_hub);
-        } else {
+
+    const onCloseReminderModal = () => {
+        setIsReminderModalOpen(false);
+    };
+
+    const onContinueReminderModal = () => {
+        createPasskey();
+        if (!error) {
             passkeysMenuActionEventTrack('create_passkey_reminder_passed');
-            createPasskey();
-            setIsModalOpen(false);
         }
+        setIsReminderModalOpen(false);
+    };
+
+    const onPasskeyMenuClick = (passkey_managing_status: TPasskeysStatus, passkey_data: TCurrentManagedPasskey) => {
+        setCurrentManagedPasskey(passkey_data);
+        setPasskeyStatus(passkey_managing_status);
     };
 
-    const onCloseRegistration = () => {
-        onCloseModal(cancelPasskeyRegistration);
+    const onPrimaryButtonClick = (passkey_data?: Partial<TCurrentManagedPasskey>) => {
+        if (
+            passkey_status === PASSKEY_STATUS_CODES.NO_PASSKEY ||
+            passkey_status === PASSKEY_STATUS_CODES.LIST ||
+            passkey_status === PASSKEY_STATUS_CODES.LEARN_MORE
+        ) {
+            const subform_name = passkey_status === PASSKEY_STATUS_CODES.LEARN_MORE ? 'passkey_info' : 'passkey_main';
+            passkeysMenuActionEventTrack('create_passkey_started', { subform_name });
+            startPasskeyRegistration();
+            setIsReminderModalOpen(true);
+        }
+        if (passkey_status === PASSKEY_STATUS_CODES.CREATED) {
+            passkeysMenuActionEventTrack('create_passkey_continue_trading');
+            history.push(routes.traders_hub);
+        }
+        if (passkey_status === PASSKEY_STATUS_CODES.RENAMING) {
+            renamePasskey(current_managed_passkey.id, passkey_data?.name ?? current_managed_passkey.name);
+        }
+        if (passkey_status === PASSKEY_STATUS_CODES.REMOVED) {
```

Comment:
NIT: can we rename "REMOVED" to "REVOKED"

Reply:
@amina-deriv , but in design it is rename, I just follow the design flow. :)